a2aclient/rest.go

```diff
@@ -161,6 +163,45 @@ func (t *RESTTransport) GetTask(ctx context.Context, query *a2a.TaskQueryParams)
 	return &task, nil
 }
 
+// ListTasks retrieves a list of tasks.
+func (t *RESTTransport) ListTasks(ctx context.Context, request *a2a.ListTasksRequest) (*a2a.ListTasksResponse, error) {
+	path := "/v1/tasks"
+
+	query := url.Values{}
+	if request.ContextID != "" {
+		query.Add("contextId", string(request.ContextID))
+	}
+	if request.Status != "" {
+		query.Add("status", string(request.Status))
+	}
+	if request.PageSize != 0 {
+		query.Add("page_size", strconv.Itoa(request.PageSize))
+	}
+	if request.PageToken != "" {
+		query.Add("page_token", string(request.PageToken))
+	}
+	if request.HistoryLength != 0 {
+		query.Add("history_length", strconv.Itoa(request.HistoryLength))
+	}
+	if request.LastUpdatedAfter != nil {
+		query.Add("last_updated_after", request.LastUpdatedAfter.Format(time.RFC3339))
+	}
+	if request.IncludeArtifacts {
+		query.Add("include_artifacts", "true")
+	}
```

Comment:
Based on [this](https://a2a-protocol.org/latest/specification/#115-query-parameter-naming-for-request-parameters) query parameters should use camelCase.

```suggestion
	if request.PageSize != 0 {
		query.Add("pageSize", strconv.Itoa(request.PageSize))
	}
	if request.PageToken != "" {
		query.Add("pageToken", string(request.PageToken))
	}
	if request.HistoryLength != 0 {
		query.Add("historyLength", strconv.Itoa(request.HistoryLength))
	}
	if request.LastUpdatedAfter != nil {
		query.Add("lastUpdatedAfter", request.LastUpdatedAfter.Format(time.RFC3339))
	}
	if request.IncludeArtifacts {
		query.Add("includeArtifacts", "true")
	}
```